Knife-throwing Mini-game V1

tessapower opened the pull request:
Merging this PR will:

- Add a new knife-throwing mini-game.
- Support basic game mechanics by left-clicking.
- Support playing 5 rounds, but does not increase the difficulty (reserved for V2)
- Support keeping track of the player's score, but currently does nothing with overall game state (reserved for V2).
- Add basic range bars with simple graphics as placeholders.

To best make use of @ZedXC's mini-game class, we should first merge #4, before rebasing and merging any changes from this branch.

Closes #13.

tessapower committed:
Add gitignore entries for VS Code

tessapower committed:
Create basic range with aim bar movement for knife-throwing game

tessapower committed:
Add functions to detect aim bar overlapping ranges

tessapower committed:
Keep track of current score, rounds, and game won/lost

tessapower committed:
Add nice comments to knife-throwing mini-game variables

tessapower committed:
Add knife-throwing to array of mini-games

tessapower committed:
Add knife-throwing to array of mini-games

ZedXC commented on jest-quest/scripts/managers/game_manager.gd:
```diff
@@ -4,7 +4,8 @@ var score = 0
+const knife_throwing = preload("res://scenes/mini_games/knife_throwing/knife_throwing.tscn")
 
 func _ready():
     mini_games.append(juggling)
```
_ready needs to append knife_throwing
` mini_games.append(knife_throwing)`

ZedXC reviewed (changes requested):
Just noticed that one thing meaning the minigame wont show up when ran in main

ZedXC committed:
Bug Fixes

- Made it so knife throwing is centered around 0,0 this way it will actually show up.
- Made it so knife throwing can be played in main

ZedXC committed:
Merge branch 'tp/knife-throwing' of https://github.com/tessapower/ggj-2024 into tp/knife-throwing

ZedXC committed:
small changes

Tabbed the game manager
Unhid previously hidden image
